File: Java/remove-duplicates-from-sorted-array.java
//  Remove Duplicates from Sorted Array

// Time O(N), Space O(1)
class Solution {
    public int removeDuplicates(int[] nums) {
        int len = 1;
        if (nums.length == 1)
            return len;
        for (int i=1; i<nums.length; i++)
            if (nums[i] != nums[len-1])
                nums[len ++] = nums[i];
        return len;
    }
}

File: Leetcode200/remove-duplicates-from-sorted-array.cc
//  Remove Duplicates from Sorted Array

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 2)    return nums.size();
        int res = 1;
        for (auto &x : nums) 
            if (x != nums[res-1])
                nums[res ++] = x;
        return res;
    }
};